### PR TITLE
adding opentelemetry autoconfigure

### PR DESCRIPTION
**Pull request by nerudadhich**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/3608

### CODE REVIEW

**Review by theletterf — APPROVED** [review]

**Comment by svrnm** [comment]
Is autoconfigure the way to go, or should we have both, or at least a note on how people can add resources "manually"? cc @open-telemetry/java-approvers

**Comment by nerudadhich** [comment]
> Is autoconfigure the way to go, or should we have both, or at least a note on how people can add resources "manually"? cc @open-telemetry/java-approvers

Hi @open-telemetry/java-approvers I am planning to build this example on auto configure with a note of manual instrumentation, Please let me know if that would be okay ?

File: content/en/docs/languages/java/exporters.md [hunk 1]
```diff
@@ -104,22 +104,7 @@ import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
 
 import io.opentelemetry.api.OpenTelemetry;
-import io.opentelemetry.api.common.Attributes;
```
Comment (jack-berg) [inline content/en/docs/languages/java/exporters.md:107]: This text above would need to be adjusted to reflect this change:

> In the case of [manual configuration] you can update the
[example app](/docs/languages/java/instrumentation#example-app) like the
following:

There are similar manual configuration snippets below for prometheus and zipkin that should be updated as well if this is the direction we want to go.

**Comment by jack-berg** [comment]
> Is autoconfigure the way to go, or should we have both, or at least a note on how people can add resources "manually"? cc @open-telemetry/java-approvers

Autoconfigure is preferred because its used by otel java agent so will be familiar to most users, and because its considerably easier than programatic config.

File: content/en/docs/languages/java/exporters.md (hunk 1 above)
Comment (nerudadhich) [reply content/en/docs/languages/java/exporters.md:107]: Yes, I have that planned. Let me make these changes and then you can review. Thanks for prompt reply.

**Comment by svrnm** [comment]
@nerudadhich please feel free to move this PR out of a draft state, this is way beyond a draft I would argue and it sometimes helps approvers&maintainers to see what to look at.

**Comment by nerudadhich** [comment]
Hi @jack-berg  @trask , I am working on Prometheus exporter using auto configure . I am able to expose the metrics but I don't see metrics of web server. What dependency I need to add for this ?

Exposed metrics
```
# TYPE otlp_exporter_exported_total counter
otlp_exporter_exported_total{otel_scope_name="io.opentelemetry.exporters.otlp-grpc",success="false",type="span"} 4.0
# TYPE otlp_exporter_seen_total counter
otlp_exporter_seen_total{otel_scope_name="io.opentelemetry.exporters.otlp-grpc",type="span"} 4.0
# HELP queueSize_ratio The number of items queued
# TYPE queueSize_ratio gauge
queueSize_ratio{otel_scope_name="io.opentelemetry.sdk.logs",processorType="BatchLogRecordProcessor"} 0.0
queueSize_ratio{otel_scope_name="io.opentelemetry.sdk.trace",processorType="BatchSpanProcessor"} 0.0
# TYPE target_info gauge
target_info{service_name="dice-server",telemetry_sdk_language="java",telemetry_sdk_name="opentelemetry",telemetry_sdk_version="1.36.0"} 1
```

dependencies 
```
dependencies {
    implementation("org.springframework.boot:spring-boot-starter-web");
    implementation("io.opentelemetry:opentelemetry-api");
    implementation("io.opentelemetry:opentelemetry-sdk");
    implementation("io.opentelemetry:opentelemetry-exporter-logging");
    implementation("io.opentelemetry:opentelemetry-exporter-otlp:1.36.0");
    implementation("io.opentelemetry:opentelemetry-exporter-prometheus:1.36.0-alpha");
    implementation("io.opentelemetry.semconv:opentelemetry-semconv:1.23.1-alpha");
    implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure");
}
```

**Comment by svrnm** [comment]
@open-telemetry/java-approvers @open-telemetry/java-instrumentation-approvers can someone help @nerudadhich with the open questions for this PR? Thanks!

**Comment by nerudadhich** [comment]
Hi @trask @jack-berg Could you please help here, it's been long time since the PR is open.

**Comment by jack-berg** [comment]
Hey sorry for the delay. If I'm ever slow to respond in the future, just ping me or send me a message on CNCF slack. I get a ton of notifications and some stuff slips through the cracks!

> Hi @jack-berg @trask , I am working on Prometheus exporter using auto configure . I am able to expose the metrics but I don't see metrics of web server. What dependency I need to add for this ?

The SDK autoconfigure module only automatically configures the SDK. There's no automatic installation of instrumentation. To install the instrumentation for spring web, you'll need to add [this dependency](https://github.com/newrelic/newrelic-opentelemetry-examples/blob/c003eb9cd0fa968365cd7ed9c0e910aa623e713b/getting-started-guides/java/instrumented/build.gradle#L29), and add [code like this](https://github.com/newrelic/newrelic-opentelemetry-examples/blob/c003eb9cd0fa968365cd7ed9c0e910aa623e713b/getting-started-guides/java/instrumented/src/main/java/com/example/demo/Application.java#L59-L62).

**Comment by svrnm** [comment]
@nerudadhich do you have everything to proceed here?

**Comment by nerudadhich** [comment]
> @nerudadhich do you have everything to proceed here?

yes @svrnm , a bit busy on some personal things. Will try to rep up next week.

**Comment by svrnm** [comment]
@nerudadhich any updates?

**Comment by svrnm** [comment]
@nerudadhich any updates?

**Comment by svrnm** [comment]
@nerudadhich since we are doing a major rework of the java docs (#4966) and this PR has not been updated for a few months now I will close it

**Pull request closed without merging**